src/utils/scanningUtils.ts

```diff
@@ -28,3 +32,76 @@ export const deleteVoucher = (
 export const testContext = (dispatch: ScanningDispatch) => {
   dispatch({ type: 'TEST' });
 };
+
+/** Declares prop types for usePreventLeave */
+type PreventLeaveProps = {
+  hasUnsavedChanges: boolean;
+  navigation: NativeStackNavigationProp<
+    ScannerStackParamList,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    any,
+    undefined
+  >;
+  dispatch: ScanningDispatch;
+};
+
+/** Event handler that prevents user from leaving screen if there are unsaved changes */
+export const handlePreventLeave = ({
+  hasUnsavedChanges,
+  navigation,
+  dispatch,
+}: PreventLeaveProps) => {
+  if (hasUnsavedChanges) {
+    // Prompt the user before leaving the screen
+    Alert.alert(
+      'Discard changes?',
+      'You have unsaved changes. Are you sure to discard them and leave the screen?',
+      [
+        {
+          text: "Don't leave",
+          style: 'cancel',
+          onPress: () => {
+            // Do nothing
+          },
+        },
+        {
+          text: 'Discard',
+          style: 'destructive',
+          // If the user confirms, then we reset the Transaction
+          onPress: () => {
+            newInvoice(dispatch);
+            navigation.navigate('VoucherEntryStartScreen');
+          },
+        },
+      ],
+    );
+  } else {
+    Alert.alert('Exit?', "Are you sure you don't want to enter any vouchers?", [
+      {
+        text: 'Cancel',
+        style: 'cancel',
+        onPress: () => {
+          // Do nothing
+        },
+      },
+      {
+        text: 'Exit',
+        style: 'destructive',
+        onPress: () => {
+          navigation.navigate('VoucherEntryStartScreen');
+        },
+      },
+    ]);
+  }
+};
+
+/** Displays a toast on successful voucher entry */
+export const showSuccessToast = () => {
+  Toast.show({
+    type: 'success',
+    position: 'top',
+    topOffset: 50,
+    text1: 'Voucher Added!',
+    visibilityTime: 2000,
+  });
+};
```

Comment:
Minor styling, but the text feels very small and may be hard to read on some devices.

Reply:
Will implement this in a future release (TODO comment added)